flink-core/src/main/java/org/apache/flink/util/Preconditions.java

```diff
@@ -234,6 +234,22 @@ public static void checkElementIndex(int index, int size) {
 		}
 	}
 
+	/**
+	 * Ensures that the given index is valid for an array, list or string of the given size.
+	 *
+	 * @param index index to check
+	 * @param size size of the array, list or string
+	 *
+	 * @throws IllegalArgumentException Thrown, if size is negative.
+	 * @throws IndexOutOfBoundsException Thrown, if the index negative or greater than or equal to size
+	 */
+	public static void checkElementIndex(int index, int size, @Nullable String errorMessage) {
+		checkArgument(size >= 0, "Size was negative.");
+		if (index < 0 || index >= size) {
+			throw new IndexOutOfBoundsException(errorMessage + " Index: " + index + ", Size: " + size);
```

Comment:
other Preconditions methods print also print "null" if no error message was given. granted, they use String.valueOf(errorMessage), but it's essentially the same.